gateway/ln-gateway/src/events.rs

```diff
@@ -22,4 +32,288 @@ pub const ALL_GATEWAY_EVENTS: [EventKind; 11] = [
     DepositConfirmed::KIND,
 ];
 
-// TODO: Add Gateway specific events
+/// Searches through the event log for all events that occurred within the
+/// specified time bounds. Because it is inefficient to search the log
+/// backwards, instead this function traverses the log forwards, but in batches.
+/// All events are appended to an array until the cutoff event where the
+/// timestamp is greater than the start timestamp or the end of the log is hit.
+pub async fn get_events_for_duration(
+    client: &Arc<ClientHandle>,
+    start: SystemTime,
+    end: SystemTime,
+) -> Vec<PersistedLogEntry> {
+    const BATCH_SIZE: u64 = 10_000;
+    let mut end_position = {
+        let mut dbtx = client.db().begin_transaction_nc().await;
+        dbtx.get_next_event_log_id().await
+    };
+
+    let mut start_position = end_position.saturating_sub(BATCH_SIZE);
```

Comment:
The mixing of `_index` and `_position` here is confusing. Should this one be `batch_end` and `batch_start`, and one below `time_bound_start` and `time_bound_end` or something?

Reply:
Yes that is better naming